main/python/cmdLineUtils.py

```diff
@@ -598,8 +598,19 @@ def copyRootObjectRecursive(sourceFile,sourcePathSplit,destFile,destPathSplit,re
     """
     retcode = 0
     replaceOption = replace
+    seen = {}
     for key in getKeyList(sourceFile,sourcePathSplit):
         objectName = key.GetName()
+
+        # write keys only if the cycle is higher than before
```

Comment:
This solves the problem.   We should document though that it also means that the copy is not 'faithful' but 'trims' the older cycle away (isn't it?)

Reply:
Yes, it will trim old cycles. I am not sure where to document that, but will gladly do it. Otherwise we have to gather all metadata first, and write out in the reverse order the objects that have multiple cycles. I decided to trim old cycles instead, since users will benefit from the space being saved. I don't know if people rely on using previous cycles or not, although it is unlikely that they do.

Reply:
The best to document rootcp is to extend the help you get when typing "rootcp --help"

Reply:
> I don't know if people rely on using previous cycles or not, although it is unlikely that they do.

We had at least one user that was storing multiple objects under the same name *intentionally*, i.e. in his case each cycle was important and not a back up copy.

That said, for rootcp, I think it is fine to do the trimming which indeed is likely to help the majority of users.